cachet-tools/purge-cachet.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+
+import requests
+import os
+import json
+
+CACHET_HOSTNAME = os.environ.get("CACHET_HOSTNAME")
+URL = f"https://{CACHET_HOSTNAME}/api/v1/components"
+HEADERS = { 
+    'X-Cachet-Token': os.environ.get("CACHET_TOKEN")
+}
+
+with requests.Session() as session:
+    session.headers.update(HEADERS)
+
+    response = session.get(URL  + "/groups", verify=False)
+
+    if (response.status_code >= 200 and response.status_code <= 299):
```

Comment:
I would invert the if clause, e.g. `if (response.status_code >= 300):` and then raise an exception or something.  That reduces the indention for all this code.  And you'll want to do the same thing for your other requests.